File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []
    for i in range(len(progresses)):
        count = 0
        while progresses[i] < 100:
            progresses[i] += speeds[i]
            count += 1
        days.append(count)
    
    current_day = days.pop(0)
    count_day = 1
    while days:
        next_day = days.pop(0)
        if current_day >= next_day:
            count_day += 1
        else:
            answer.append(count_day)
            count_day = 1
            current_day = next_day
    answer.append(count_day)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Deque<Integer> answer = new ArrayDeque<>();
        Deque<Integer> days = new ArrayDeque<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int day = 0;
            while (progresses[i] < 100) {
                progresses[i] += speeds[i];
                day += 1;
            }
            days.add(day);
        }
        
        int cur_day = days.poll();
        int count = 1;
        
        while (!days.isEmpty()) {
            int next_day = days.poll();
            
            if (cur_day >= next_day) {
                count++;
            } else {
                answer.add(count);
                count = 1;
                cur_day = next_day;
            }
        }
        
        answer.add(count);
        
        int[] ans = new int[answer.size()];
        int index = 0;
        for (int num: answer) {
            ans[index++] = num;
        }
        
        return ans;
    }
}